.github/workflows/publish-spark-metadata-client.yaml

```diff
@@ -3,8 +3,16 @@ on:
   workflow_dispatch:
 jobs:
   publish-spark-metadata-client:
-    name: Publish Spark metadata client
+    name: Publish Spark metadata client - hadoop version ${{ matrix.hadoop-version }}
     runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - hadoop-version: "3.2.1"
+            artifact-version: "0.15.1-hadoop-3.2.1"
```

Comment:
So, this pattern isn't compatible with semver... The version should end with the semver variables, i.e. <major>.<minor>.<patch>, and not with the hadoop version (or anything else) since that basically means that the version is not a production version